cores/esp32/HardwareSerial.cpp

```diff
@@ -114,37 +114,46 @@ void serialEventRun(void)
 }
 #endif
 
-
 HardwareSerial::HardwareSerial(int uart_nr) : _uart_nr(uart_nr), _uart(NULL), _rxBufferSize(256) {}
 
+enum { UART_NUM_0, UART_NUM_1, UART_NUM_2 };
+
 void HardwareSerial::begin(unsigned long baud, uint32_t config, int8_t rxPin, int8_t txPin, bool invert, unsigned long timeout_ms, uint8_t rxfifo_full_thrhd)
 {
     if(0 > _uart_nr || _uart_nr >= SOC_UART_NUM) {
         log_e("Serial number is invalid, please use numers from 0 to %u", SOC_UART_NUM - 1);
         return;
     }
+
+    // First Time or after end() --> set default Pins
+    if (!uartIsDriverInstalled(_uart)) {
+        switch (_uart_nr) {
+            case UART_NUM_0:
+                rxPin = rxPin < 0 ? SOC_RX0 : rxPin;
+                txPin = txPin < 0 ? SOC_TX0 : txPin;
+            break;
+#if SOC_UART_NUM > 1                   // may save some flash bytes...
+            case UART_NUM_1:
+                rxPin = rxPin < 0 ? RX1 : rxPin;
+                txPin = txPin < 0 ? TX1 : txPin;
+            break;
+#endif
+#if SOC_UART_NUM > 2                   // may save some flash bytes...
+            case UART_NUM_2:
+                rxPin = rxPin < 0 ? RX2 : rxPin;
+                txPin = txPin < 0 ? TX2 : txPin;
+            break;
+#endif
```

Comment:
always add `default:` for switch statements. ESP-IDF will complain

Reply:
done.